src/models/WASZcapStorage.ts

```diff
@@ -0,0 +1,103 @@
+// @ts-nocheck
+import { ZcapClient } from '@digitalcredentials/ezcap';
+import { Ed25519VerificationKey2020 } from '@digitalcredentials/ed25519-verification-key-2020';
+import { Ed25519Signature2020 } from '@digitalcredentials/ed25519-signature-2020';
+
+export interface WasZcapStorageConfig {
+	appInstance: {
+		publicKeyMultibase: string;
+		privateKeyMultibase: string;
+		controller: string;
+		id: string;
+	};
+	capability: {
+		"@context": string[];
+		allowedAction: string[];
+		controller: string;
+		expires: string;
+		id: string;
+		invocationTarget: string;
+		parentCapability: string;
+		proof: {
+			capabilityChain: string[];
+			type: string;
+			created: string;
+			proofPurpose: string;
+			proofValue: string;
+			verificationMethod: string;
+		};
+	};
+}
+
+export class WASZcapStorage {
+	private zcapClient: any;
+	private capability: any;
+    private ready: Promise<void>;
+
+	constructor(config: WasZcapStorageConfig) {
+		if (!config?.appInstance?.publicKeyMultibase || !config?.appInstance?.privateKeyMultibase) {
+			throw new Error('appInstance is missing key material');
+		}
+		if (!config?.capability) {
+			throw new Error('capability (zcap) is required');
+		}
+		this.capability = config.capability;
+        this.ready = this.initClient(config.appInstance);
+	}
+
+    private async initClient(appInstance: any) {
+        const key = await Ed25519VerificationKey2020.from(appInstance);
+        const signer = key.signer();
+        // ezcap expects invocationSigner.id
+        (signer as any).id = key.id;
```

Comment:
This is already done automatically in the `.signer()` function -- see https://github.com/digitalcredentials/ed25519-verification-key-2020/blob/main/src/Ed25519VerificationKey2020.ts#L582
```suggestion
```